x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -118,10 +118,25 @@ export function setLayerErrorStatus(id, errorMessage) {
 }
 
 export function toggleLayerVisible(layerId) {
-  return {
-    type: TOGGLE_LAYER_VISIBLE,
-    layerId
+  return async (dispatch, getState) => {
+    //if the current-state is invisible, we also want to sync data
+    //e.g. if a layer was invisible at start-up, it won't have any data loaded
+    const layer = getLayerList(getState()).find(layer => {
+      return layerId === layer.getId();
+    });
+    if (!layer) {
+      return;
+    }
+    const makeVisible = !layer.isVisible();
```

Comment:
I do not get why makeVisibible is the inverse of layer.isVisible(). I agree that `toggleLayerVisible` needs to dispatch `syncDataForLayer` because layers that were invisible may need to sync data. Why not this do this instead and only sync data when the layer is made visible?

```
export function toggleLayerVisible(layerId) {
  return async (dispatch, getState) => {
    const layer = getLayerList(getState()).find(layer => {
      return layerId === layer.getId();
    });
    if (!layer) {
      return;
    }
    await dispatch({
      type: TOGGLE_LAYER_VISIBLE,
      layerId
    });
    if (layer.isVisible()) {
      dispatch(syncDataForLayer(layerId));
    }
  };
}
```

Reply:
> I do not get why makeVisibible is the inverse of layer.isVisible().

Because it was a toggle action, the action should be setting the layer to the opposite of what it was before. Personally I don't love toggles and would prefer to pass an explicit boolean value from the onclick handler fwiw.